pandas/io/parsers.py

```diff
@@ -1791,19 +1791,17 @@ def __init__(self, f, **kwds):
         self.comment = kwds['comment']
         self._comment_lines = []
 
-        if isinstance(f, compat.string_types):
-            f = _get_handle(f, 'r', encoding=self.encoding,
-                            compression=self.compression,
-                            memory_map=self.memory_map)
-            self.handles.append(f)
-        elif self.compression:
-            f = _wrap_compressed(f, self.compression, self.encoding)
-            self.handles.append(f)
-        # in Python 3, convert BytesIO or fileobjects passed with an encoding
-        elif compat.PY3 and isinstance(f, compat.BytesIO):
-            from io import TextIOWrapper
+        add_handle = (
```

Comment:
I think `add_handle` is only used here -- I created it for this PR, so nowhere else depends on it.

I'm still a little unsure about the purpose of `self.handles` -- should it only contain file handles because right now we're adding handles that aren't files (for example in the case of `compat.PY3 and isinstance(f, compat.BytesIO)`)?